src/backend/distributed/connection/connection_management.c

```diff
@@ -160,6 +160,41 @@ AfterXactConnectionHandling(bool isCommit)
 }
 
 
+/*
+ * AnyConnectionAccessedPlacements goes over all the connections in the ConnectionHash
+ * and returns true if any of the connection has any referenced placements. This is
+ * useful to detect whether we're in a distirbuted transaction and already executed
+ * at least one command that accessed to a placement.
+ */
+bool
+AnyConnectionAccessedPlacements(void)
```

Comment:
this function is potentially quite expensive, is there any faster way we can determine this?

Reply:
I guess the following should work:  `return hash_get_num_entries(ConnectionPlacementHash) > 0;`